### PR TITLE
feat(nimbus): Update mozilla-nimbus-schemas and pydantic

### LINKED ISSUE #11547: Update Experimenter to Pydantic v2
To use the new version of mozilla-nimbus-schemas, we need to also update Experimenter to Pydantic v2.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4875)

### PR DESCRIPTION
**Pull request by brennie**
Because:

- mozilla-nimbus-schemas is on pydantic v2; and
- experimenter is not

this commit:

- updates experimenter to use pydantic v2;
- updates experimenter to use the new version of mozilla-nimbus-schemas.

Fixes #11547

### CODE REVIEW

**Comment by brennie** [comment]
This had to be reverted for causing prod failures but we couldnt reproduce this locally. Maybe we add more logging or something and re-push this, hoping to see the error again with more context?

File: experimenter/experimenter/features/__init__.py [hunk 1]
```diff
@@ -34,16 +35,17 @@
 class Feature:
     slug: str
     application_slug: str
-    model: Union[FeatureWithExposure, FeatureWithoutExposure]
-    version: Optional[Version] = None
+    model: SdkFeature | DesktopFeature
```
Comment (jaredlockhart) [inline experimenter/experimenter/features/__init__.py:38]: I like that 🙏

File: experimenter/experimenter/features/tests/test_features.py [hunk 2]
```diff
@@ -39,29 +40,30 @@ def test_load_all_features(self):
             Feature(
                 slug="someFeature",
                 application_slug="firefox-desktop",
-                model=FeatureWithExposure.parse_obj(
+                model=DesktopFeature.parse_obj(
```
Comment (jaredlockhart) [inline experimenter/experimenter/features/tests/test_features.py:43]: Ah that's much nicer 🙏

File: experimenter/experimenter/features/tests/test_features.py [hunk 3]
```diff
@@ -76,13 +78,13 @@ def test_load_all_features(self):
             Feature(
                 slug="defaultBrowser",
                 application_slug="fenix",
-                model=FeatureWithoutExposure.parse_obj(
+                model=SdkFeature.parse_obj(
```
Comment (jaredlockhart) [inline experimenter/experimenter/features/tests/test_features.py:81]: Fantastic

**Review by jaredlockhart — APPROVED** [review]
Awesome.  Big ugly migration but everything reads much nicer now, thnx for taking this on @brennie 🙏 🎉

**Pull request merged**